### PR TITLE
Add articles schema and CRUD routes

### PR DESCRIPTION
**Pull request by tqian1**
This change adds the basic mongoose schema for articles, the routes for navigating to its views, as well as a couple CRUD endpoints to go with it. It also disables CSRF checking for /api/articles.

The articles route exists at /articles
The JSON api for articles is access at /api/articles

### CODE REVIEW

File: models/Article.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const mongoose = require('mongoose');
+
+const articleSchema = new mongoose.Schema({
+  title: { type: String, unique: true },
```
Comment (wuweiweiwu) [inline models/Article.js]: I think we'll need author too

File: models/Article.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  status: Number,
+  description: String,
+  category: Number
```
Comment (wuweiweiwu) [inline models/Article.js]: how should we map `category` to actual categories? client side?

Comment (tqian1) [reply models/Article.js]: Will we be using an enumerated list of categories like how status is usually implemented? or will it have its own schema. If Categories will be its own type of object in the DB then i can link the two together, the usual fetch will include the id reference, or i can make it fetch with a deep populate.

File: models/Article.js (hunk 1 above)
Comment (tqian1) [reply models/Article.js]: I will update this to include author, as well as store enumeration for status somewhere.

File: app.js [hunk 3]
```diff
@@ -86,7 +87,7 @@ app.use(passport.initialize());
 app.use(passport.session());
 app.use(flash());
 app.use((req, res, next) => {
-  if (req.path === '/api/upload') {
+  if (req.path === '/api/upload' || '/api/article') {
```
Comment (tqian1) [inline app.js]: this is how to remove certain paths from csrf, i have removed the api level for article.

File: controllers/api.js [hunk 4]
```diff
@@ -25,6 +26,31 @@ const { Venues, Users } = require('node-foursquare')({
   }
 });
 
+/**
+ * GET /api/articles
+ * List of API examples.
+ */
+exports.getArticles = (req, res) => {
+  Article.find((err, articles) => {
+    res.send(articles);
```
Comment (wuweiweiwu) [inline controllers/api.js]: We should have error checking

File: controllers/api.js [hunk 5]
```diff
@@ -25,6 +26,31 @@ const { Venues, Users } = require('node-foursquare')({
...
+/**
+ * POST /api/articles
+ * List of API examples.
+ */
+exports.postArticles = (req, res) => {
+  var article = new Article(req.body);
+
+  article.save((err, article) => {
+    if (err)
+      res.send(err);
```
Comment (tqian1) [inline controllers/api.js]: 500 responses here

File: controllers/api.js (hunk 4 above)
Comment (tqian1) [inline controllers/api.js]: pagination

**Pull request merged**